### ISSUE TITLE
search for board specific shield overlays doesn't always work

### ISSUE THREAD

**Issue by galak** [zephyrproject-rtos/zephyr#19078]
If the shield name doesn't match the dir name we don't look in correct place.  For example shield `x_nucleo_iks01a3_shub`.

**Comment by galak** [comment]
So we look here for things:

```
look for /home/galak/git/zephyr/boards/shields/x_nucleo_iks01a3_shub/boards/stm32mp157c_dk2.overlay
look for /home/galak/git/zephyr/boards/shields/x_nucleo_iks01a3_shub/x_nucleo_iks01a3_shub.conf
look for /home/galak/git/zephyr/boards/shields/x_nucleo_iks01a3_shub/boards/stm32mp157c_dk2.conf
```

The dir is incorrect, however not quite sure how we correctly handle this for multiple different shields in on dir like `x_nucleo_iks01a3` has.

**Comment by erwango** [comment]
Shouldn't we split into 2 different shields? 
From what I understand the different modes enables quite different behavior and that's if we had 2 different shields finally.
@avisconti, what do you think ?

**Comment by mike-scott** [comment]
I have to agree here.  When you enable the sensorhub mode (which is a jumper) it completely changes the functionality of the shield.
